src/components/entity/ha-state-label-badge.html

```diff
@@ -47,120 +47,115 @@
+        case 'binary_sensor':
+        case 'device_tracker':
+        case 'updater':
+        case 'sun':
+        case 'alarm_control_panel':
+          return null;
+        case 'sensor':
+        default:
+          return state.state === 'unknown' ? '-' : state.state;
+      }
     }
-    var domain = window.hassUtil.computeDomain(state);
-    switch (domain) {
-      case 'alarm_control_panel':
-        if (state.state === 'pending') {
-          return 'mdi:clock-fast';
-        } else if (state.state === 'armed_away') {
-          return 'mdi:nature';
-        } else if (state.state === 'armed_home') {
-          return 'mdi:home-variant';
-        } else if (state.state === 'triggered') {
-          return 'mdi:alert-circle';
-        }
-        // state == 'disarmed'
-        return window.hassUtil.domainIcon(domain, state.state);
-      case 'binary_sensor':
-      case 'device_tracker':
-      case 'updater':
-        return window.hassUtil.stateIcon(state);
-      case 'sun':
-        return state.state === 'above_horizon' ?
-          window.hassUtil.domainIcon(domain) : 'mdi:brightness-3';
-      default:
+
+    computeIcon(state) {
+      if (state.state === 'unavailable') {
         return null;
+      }
+      var domain = window.hassUtil.computeDomain(state);
+      switch (domain) {
+        case 'alarm_control_panel':
+          if (state.state === 'pending') {
+            return 'mdi:clock-fast';
+          } else if (state.state === 'armed_away') {
+            return 'mdi:nature';
+          } else if (state.state === 'armed_home') {
+            return 'mdi:home-variant';
+          } else if (state.state === 'triggered') {
+            return 'mdi:alert-circle';
+          }
+          // state == 'disarmed'
+          return window.hassUtil.domainIcon(domain, state.state);
+        case 'binary_sensor':
+        case 'device_tracker':
+        case 'updater':
+          return window.hassUtil.stateIcon(state);
+        case 'sun':
+          return state.state === 'above_horizon' ?
+            window.hassUtil.domainIcon(domain) : 'mdi:brightness-3';
+        default:
+          return null;
+      }
     }
-  }
 
-  computeImage(state) {
-    return state.attributes.entity_picture || null;
-  }
+    computeImage(state) {
+      return state.attributes.entity_picture || null;
+    }
 
-  computeLabel(state) {
-    if (state.state === 'unavailable') {
-      return 'unavai';
+    computeLabel(state) {
+      if (state.state === 'unavailable' ||
+          STATE_LABEL_DOMAINS.includes(window.hassUtil.computeDomain(state))) {
+        // Localize the state with a special state_badge namespace, which has variations of
+        // the state translations that are truncated to fit within the badge label. Translations
+        // are only added for domains in STATE_LABEL_DOMAINS.
+        return this.localize('state_badge', state.state);
+      }
+      return state.attributes.unit_of_measurement || null;
```

Comment:
Could be done in a separate PR, but we need to translate the units too.

Reply:
Yep, we'll get there at some point 😄